### ISSUE TITLE
[BUG] Zero momentum will disable weight decay in SGD optimizer implementation

### ISSUE THREAD

**Issue by PRESIDENT810** [ml-explore/mlx#440]
**Describe the bug**
In mlx's SGD implementation:

```
    def apply_single(
        self, gradient: mx.array, parameter: mx.array, state: OptimizerState
    ):
        """Performs the SGD parameter update and stores :math:`v` in the
        optimizer state."""
        if self.momentum <= 0:
            return parameter - self.learning_rate * gradient

        v = state.get("v", mx.zeros_like(gradient))

        if self.weight_decay != 0:
            gradient += self.weight_decay * parameter

        v = self.momentum * v
        if self.dampening > 0:
            v += (1 - self.dampening) * gradient
        else:
            v += gradient

        if self.nesterov:
            update = gradient + self.momentum * v
        else:
            update = v
        state["v"] = v
        return parameter - self.learning_rate * update
```

If I don't set momentum, by default it is 0.0, so in 
```
        if self.momentum <= 0:
            return parameter - self.learning_rate * gradient
```
It will return immediately. If I set a weight decay without setting momentum, it essentially disables my weight decay because that line of code will not be reached.

From my understanding, momentum shouldn't conflict with weight decay, as in PyTorch's implementation: https://pytorch.org/docs/stable/generated/torch.optim.SGD.html#sgd, even if I don't set momentum, weight decay should still take effect.

**To Reproduce**
I'm learning this book "Dive into deep learning" and trying to implement its code using mlx, and I found this bug when implementing Linear regression
https://github.com/PRESIDENT810/d2l-en/blob/6cb8059e0ded2675871d7ae4d4ececfcd49ab25b/mlx_code/Chapter%203/3.7.ipynb

If I don't set momentum, as you can see the evaluation loss doesn't change much.
<img width="844" alt="image" src="https://github.com/ml-explore/mlx/assets/44538064/c740115e-fce5-4ac9-b565-a6e88ae52181">

However, even if I set a very small positive momentum, evaluation decreases:
<img width="837" alt="image" src="https://github.com/ml-explore/mlx/assets/44538064/48c0d257-b9c0-49fe-9d91-0e5aac2f1fcb">

**Expected behavior**
Even if there is no momentum, weight decay still take effect and prevent model from overfitting.

**Desktop (please complete the following information):**
 - OS Version: [e.g. MacOS 14.1.2]
 - Version [e.g. 0.7.0]

**Additional context**
I think this should be an easy fix. Feel free to assign it to me.

**Comment by awni** [comment]
Definitely a bug, and should be an easy fix! Thank you for volunteering to fix it.
